CMakeLists.txt

```diff
@@ -401,8 +415,22 @@ if(EMSCRIPTEN)
     xeus_cpp_set_kernel_options(xcpp)
     xeus_wasm_compile_options(xcpp)
     xeus_wasm_link_options(xcpp "web,worker")
+    target_link_options(xcpp PUBLIC
+        -sEXPORTED_RUNTIME_METHODS=FS,PATH,ERRNO_CODES
+        # add sysroot location here
+        --preload-file ${SYSROOT_PATH}/include@/include
+    )
+    # TODO: Emscripten supports preloading files just once before it generates
+    # the xcpp.data file (containing the binary representation of the file(s) we
+    # want to include in our application).
+    # Hence although we are adding supporting for Standard Headers, Libraries etc
+    # through emscripten's sysroot for now, we need to do the following:
+    # 1) Enable CppInterOp to provide us with a resource dir.
+    # 2) If the above cannot be done, we can use the resource dir provided
+    #    by llvm on emscripten-forge but would involve adding a dependency.
+    # 3) Shift the resource dir and the sysroot to a common location.
+    # 4) Preload everything required together.
```

Comment:
TODO with respect to preloading raised here. Please let me know of any questions you'll have or this needs to be rephrased.

Reply:
Technically speaking in the long run we would have a better method to load files (and libraries like clad, xsimd etc) dynamically using `mambajs`(https://github.com/emscripten-forge/mambajs) that is currently in the works.